cosmo_tester/test_suites/image_based_tests/component_test.py

```diff
@@ -0,0 +1,170 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+
+import retrying
+
+from cosmo_tester.framework import util
+from cosmo_tester.framework.examples import get_example_deployment
+
+
+def _infra(image_based_manager, ssh_key, logger, tenant, test_config):
+    example = get_example_deployment(
+        image_based_manager, ssh_key, logger, tenant, test_config,
+        upload_plugin=False)
+
+    example.blueprint_file = util.get_resource_path(
+        'blueprints/component/fake_infra.yaml'
+    )
+    example.blueprint_id = 'infra'
+    example.deployment_id = 'infra'
+    example.inputs = {}
+    return example
+
+
+def _app(image_based_manager, ssh_key, logger, tenant, test_config):
+    example = get_example_deployment(
+        image_based_manager, ssh_key, logger, tenant, test_config,
+        upload_plugin=False)
+
+    example.blueprint_file = util.get_resource_path(
+        'blueprints/component/component.yaml'
+    )
+    example.blueprint_id = 'app'
+    example.deployment_id = 'app'
+    example.inputs['agent_user'] = image_based_manager.username
+    example.create_secret = False   # don't try to create it twice
+    return example
+
+
+def _nesting_app(image_based_manager, ssh_key, logger, tenant, test_config):
+    example = get_example_deployment(
+        image_based_manager, ssh_key, logger, tenant, test_config,
+        upload_plugin=False)
+
+    example.blueprint_file = util.get_resource_path(
+        'blueprints/component/nesting_component.yaml'
+    )
+    example.blueprint_id = 'nesting_app'
+    example.deployment_id = 'nesting_app'
+    example.inputs['agent_user'] = image_based_manager.username
+    example.create_secret = False   # don't try to create it twice
+    return example
+
+
+def test_component(image_based_manager, ssh_key, logger, test_config):
+    tenant = 'test_component'
+    infra = _infra(image_based_manager, ssh_key, logger, tenant, test_config)
+    app = _app(image_based_manager, ssh_key, logger, tenant, test_config)
+
+    # We're uploading a blueprint that creates an infrastructure for a VM,
+    # and then exposes capabilities, which will be used in the application
+    logger.info('Uploading infrastructure blueprint.')
+    infra.upload_blueprint()
+    logger.info('Deploying parent component blueprint.')
+    app.upload_and_verify_install()
+
+    with util.set_client_tenant(app.manager.client, tenant):
+        # verify both deployments are installed
+        deployments = app.manager.client.deployments.list()
```

Comment:
You can probably fold these two lines into one- assert len(app.manager.client.deployments.list()) == 2?